Tools/CISupport/ews-build/steps.py

```diff
@@ -3611,6 +3611,22 @@ def start(self):
 
         self.setCommand(self.command + customBuildFlag(self.getProperty('platform'), self.getProperty('fullPlatform')))
         self.command.extend(self.command_extra)
+
+        self.command = self.shell_command(' '.join(self.command) + ' 2>&1 | python3 Tools/Scripts/filter-jsc-tests')
+
+        steps_to_add = [
+            GenerateS3URL(
+                f"{self.getProperty('fullPlatform')}-{self.getProperty('architecture')}-{self.getProperty('configuration')}-{self.name}",
+                extension='txt',
+                content_type='text/plain',
+            ), UploadFileToS3(
+                'logs.txt',
+                links={self.name: 'Full logs'},
+                content_type='text/plain',
+            )
+        ]
+        self.build.addStepsAfterCurrentStep(steps_to_add)
```

Comment:
I tested it on UAT instance and found this this doesn't really work.
This is because of code in evaluateCommand() method in this case.

In case of success we call self.build.buildFinished() which finish the build immediately without calling the steps for uploading the build logs.
In case of failure, it does another self.build.addStepsAfterCurrentStep() call which calls those steps before these steps. (see https://github.com/buildbot/buildbot/issues/4874).

Better to move this code inside evaluateCommand() method and ensure there is single call to self.build.addStepsAfterCurrentStep.

Note that one difference in this class as compared to build.webkit.org is that from build.webkit.org uses new-style run method while this code uses old-style start() method.

Reply:
Thank you for explaining! Made the fix.